src/types.db

```diff
@@ -57,7 +57,13 @@ df_inodes               value:GAUGE:0:U
 dilution_of_precision   value:GAUGE:0:U
 disk_allocation         value:GAUGE:0:U
 disk_capacity           value:GAUGE:0:U
+disk_discard_complete   value:DERIVE:0:U
+disk_discard_merged     value:DERIVE:0:U
+disk_discard_sectors    value:DERIVE:0:U
+disk_discard_time       value:DERIVE:0:U
 disk_error              value:GAUGE:0:U
+disk_flush_complete     value:DERIVE:0:U
+disk_flush_time         value:DERIVE:0:U
```

Comment:
Oh, I didn't realize that these are actually different type metrics. Putting the read and write metrics into one "type", e.g. for `disk_ops`, bites us again :(

I see two options:

1.  Just add new metrics for these fields, like the PR currently does.

    If we go down the route, I'd like to rename `…_complete` to `…_ops` though, to be consistent with other disk metrics.
1.  Change existing metrics to use single-value types. Because this is a breaking change, it must be guarded by a config option that's off by default. In this option, we'd use types like `disk_ops_complex` and set the `type_instance` to "read", "write", "discard", "flush".

Overall I think that 1. is less disruptive to users. What do you think?

Reply:
IMHO, I guess we should first let those new metrics align with old ones, it might be an easier way to use those metrics for now, like what Option 1 does.

Option 2, if you mean we change things into e.g.  "type: 'disk_ops', type_instance: 'read'", "type: 'disk_flush', type_instance: 'ops'"? If I understand correctly, this also sounds reasonable to me.  
But since we need a new conf option to implement this, maybe it is inconvenient to only let new metrics work on new options (which makes users confused I guess). 

So maybe we should go on this PR for Option 1 to let it work first. Then start a new PR for adding a conf option to let `disk_ops_complex` with `type_instance` filled to substitute existing metrics if necessary.  And it is okay for me to try working on this (Option 2). ;)

Reply:
SGTM, thanks for making those changes :D